src/core/api/thread_api.cpp

```diff
@@ -501,4 +501,20 @@ bool otThreadIsAnycastLocateInProgress(otInstance *aInstance)
 }
 #endif
 
+void otThreadLeaveNetwork(otInstance *aInstance)
+{
+    static constexpr otOperationalDatasetTlvs kEmptyDatasetTlvs = {};
+
+    SuccessOrAssert(otThreadSetEnabled(aInstance, false));
+    if (otIp6IsEnabled(aInstance))
+    {
+        SuccessOrAssert(otIp6SetEnabled(aInstance, false));
+    }
+    SuccessOrAssert(otDatasetSetActiveTlvs(aInstance, &kEmptyDatasetTlvs));
+    SuccessOrAssert(otDatasetSetPendingTlvs(aInstance, &kEmptyDatasetTlvs));
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+    SuccessOrAssert(AsCoreType(aInstance).Get<BorderRouter::RoutingManager>().HandleLeavingNetwork());
```

Comment:
May add a otThreadForgetNetwork() which can only be called when Thread is ready, that will erase all Network related settings?